src/components/root/createModals/CreateEnvironmentModal.tsx

```diff
@@ -0,0 +1,98 @@
+import {
+	Autocomplete,
+	Button,
+	DialogActions,
+	DialogContent,
+	DialogTitle,
+	Divider,
+	FormControl,
+	FormLabel,
+	Input,
+	Modal,
+	ModalDialog,
+} from '@mui/joy';
+import { CreateModalsProps } from './createModalsProps';
+import { Environment, iconFromTabType } from '../../../types/application-data/application-data';
+import { useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { selectEnvironments } from '../../../state/active/selectors';
+import { useAppDispatch } from '../../../state/store';
+import { addNewEnvironment } from '../../../state/active/thunks/environments';
+import { addTabs, setSelectedTab } from '../../../state/tabs/slice';
+
+export function CreateEnvironmentModal({ open, closeFunc }: CreateModalsProps) {
+	const [envName, setEnvName] = useState('');
+	const [cloneFrom, setCloneFrom] = useState<string | null>(null);
+	const allEnvironments = useSelector(selectEnvironments);
+	const dispatch = useAppDispatch();
+	const createEnvironmentFunction = async () => {
+		let newEnvironment: Partial<Environment> = { __name: envName };
+		if (cloneFrom != null) {
+			const environmentToCloneFrom = allEnvironments[cloneFrom];
+			if (environmentToCloneFrom != undefined) {
+				newEnvironment = { ...structuredClone(environmentToCloneFrom), ...newEnvironment };
+				delete newEnvironment['__id'];
+			}
+		}
+		const createdEnvironmentId = await dispatch(addNewEnvironment({ data: newEnvironment })).unwrap();
+		dispatch(addTabs({ [createdEnvironmentId]: 'environment' }));
+		dispatch(setSelectedTab(createdEnvironmentId));
+	};
+	const envNameValid = envName.length > 0;
```

Comment:
I feel like this might be premature optimization (the useMemo). I think the overhead of useMemo greatly exceeds the envName.length > 0 check.

Reply:
that was added before I realized how simple the validity check would be. Will remove

Reply:
Wait I'm a bit confused, I'm not sure what allFieldsValid is even doing. It looks like it's just returning the bool calculated immediately above?

Reply:
It looks like I replied to you but I did not, github did not show me your message.